conans/model/build_info.py

```diff
@@ -1,224 +1,572 @@
 import os
 from collections import OrderedDict
+from copy import deepcopy
 
 import deprecation
 
+from conans.errors import ConanException
+from conans.model.build_info_components import Component, DepComponent
+
 DEFAULT_INCLUDE = "include"
 DEFAULT_LIB = "lib"
 DEFAULT_BIN = "bin"
 DEFAULT_RES = "res"
 DEFAULT_SHARE = "share"
+DEFAULT_BUILD = ""
 
 
-class _CppInfo(object):
```

Comment:
Same thing as described in https://github.com/conan-io/conan/pull/5408/files#r299105226

This ``_CppInfo`` object was the base class for all the object and lead to interface issues. For example, in a `CppInfo` object, a user may want to add new libraries in a recipe like this ``self.cpp_info.libs = ["my_lib"]`` while they would like to consume the information in a generator using ``self.deps_cpp_info.libs``.

While this could be done with inheritance without issue, note that the user could overwrite these values by mistake in the generator using ``self.deps_cpp_info.libs = ["ups"]. This was an issue already there before Components.

However, the real issue is when you need to compound these values from components. The implementation in this PR achieves this through a `libs` property, and this collides directly with the `libs` attribute of the public interface of `CppInfo`. So I thought keeping this inheritance structure is difficult to keep and makes it more difficult to understand the code.